package.json

```diff
@@ -2231,39 +2231,17 @@
                     "default": 10,
                     "description": "%vscode-docker.config.docker.truncateMaxLength%"
                 },
-                "docker.dockerodeOptions": {
+                "docker.environment": {
                     "type": "object",
-                    "description": "%vscode-docker.config.docker.dockerodeOptions%"
-                },
-                "docker.host": {
-                    "type": "string",
-                    "default": "",
-                    "description": "%vscode-docker.config.docker.host%",
-                    "scope": "machine-overridable"
-                },
-                "docker.context": {
-                    "type": "string",
-                    "default": "",
-                    "description": "%vscode-docker.config.docker.context%",
-                    "scope": "machine-overridable"
-                },
-                "docker.certPath": {
-                    "type": "string",
-                    "default": "",
-                    "description": "%vscode-docker.config.docker.certPath%",
-                    "scope": "machine-overridable"
-                },
-                "docker.tlsVerify": {
-                    "type": "string",
-                    "default": "",
-                    "description": "%vscode-docker.config.docker.tlsVerify%",
+                    "additionalProperties": {
+                        "type": "string"
+                    },
+                    "description": "%vscode-docker.config.docker.environment%",
                     "scope": "machine-overridable"
                 },
-                "docker.machineName": {
-                    "type": "string",
-                    "default": "",
-                    "description": "%vscode-docker.config.docker.machineName%",
-                    "scope": "machine-overridable"
+                "docker.dockerodeOptions": {
```

Comment:
`docker.dockerodeOptions` is unchanged (for now, will go away with #3263), but the diff is being weird.